pkg/operator/vspherecontroller/checks/check_error.go

```diff
@@ -162,11 +189,62 @@ func CheckClusterStatus(result ClusterCheckResult, apiDependencies KubeAPIInterf
 
 	case CheckActionBlockUpgrade:
 		return ClusterCheckBlockUpgrade, result
-
+	case CheckActionRequiresAdminAck:
+		return checkForIntreePluginUse(result, apiDependencies)
 	case CheckActionBlockUpgradeDriverInstall:
 		return ClusterCheckBlockUpgradeDriverInstall, result
-
 	default:
 		return ClusterCheckAllGood, result
 	}
 }
+
+// returns false is migration is enabled in the cluster.
+// returns true if migration is not enabled in the cluster and cluster is using
+// in-tree vSphere volumes.
+func checkForIntreePluginUse(result ClusterCheckResult, apiDependencies KubeAPIInterface) (ClusterCheckStatus, ClusterCheckResult) {
+	allGoodCheckResult := MakeClusterCheckResultPass()
+
+	pvs, err := apiDependencies.ListPersistentVolumes()
+	if err != nil {
+		reason := fmt.Errorf("vsphere csi driver installed failed with %s, unable to list pvs: %v", result.Reason, err)
```

Comment:
This error is IMO not related to CSI driver installation
(this applies also to the original PR, so I don't mind if we leave it as it is...)